22nd-jan/cacheFunction.js

```diff
@@ -0,0 +1,24 @@
+Utils = {
+  memoize: function (param) {
+    if (param in this.obj.cache) {
+      return this.obj.cache[param];
+    }
+    else {
+      let result = this.obj.fn(param);
+      this.obj.cache[param] = result;
+      return result;
+    }
+  }
+
+}
+
+function cacheFunction(functionToCache) {
+  // TODO: Need improvement
+  obj = Object.create(Utils);
```

Comment:
Creating a prototype is not required. You could also just implement the memoize as an inner function.